### PR TITLE
Add address relay/processed/rate-limited fields to peer details

### PR DESCRIPTION
**Pull request by jonatack**
This pull adds the following address fields in rpc getpeerinfo and cli -netinfo to the gui peers details:

- Address Relay (Yes/No)
- Addresses Processed (integer)
- Addresses Rate-Limited (integer)

and uses the additional horizontal space to display "Last Transaction" (instead of "Last Tx").

Screenshot from 2022-01-21 00-05-49

### CODE REVIEW

**Review by w0xlt — APPROVED** [review]
tACK 5efd391 on Ubuntu 21.10 (Qt 5.15.2).

File: src/qt/forms/debugwindow.ui [hunk 1]
```diff
@@ -1592,6 +1592,84 @@
                 </widget>
                </item>
                <item row="23" column="0">
+                <widget class="QLabel" name="peerAddrRelayEnabledLabel">
+                 <property name="toolTip">
+                  <string>Whether we relay addresses to this peer.</string>
```
Comment (hebasto) [inline src/qt/forms/debugwindow.ui:1597]: Could you add an `extracomment` attribute as a translator comment for newly added tooltip strings?

As an example see dbde0558ce73db4c901dbaa2eddc634701fa1d0d.

Comment (jonatack) [reply src/qt/forms/debugwindow.ui]: Thanks -- done.

**Comment by jonatack** [comment]
Thanks @w0xlt and @hebasto, updated.

<details><summary><code>git diff a00bc71 9fbd1bb</code></summary><p>

```diff
diff --git a/src/qt/forms/debugwindow.ui b/src/qt/forms/debugwindow.ui
index 174954087..219680102 100644
--- a/src/qt/forms/debugwindow.ui
+++ b/src/qt/forms/debugwindow.ui
@@ -1355,7 +1355,7 @@
                <item row="13" column="0">
                 <widget class="QLabel" name="peerLastTxLabel">
                  <property name="toolTip">
-                  <string>Elapsed time since a novel transaction accepted into our mempool was received from this peer.</string>
+                  <string extracomment="Tooltip text for the Last Transaction field in the peer details area.">Elapsed time since a novel transaction accepted into our mempool was received from this peer.</string>
                  </property>
                  <property name="text">
                   <string>Last Transaction</string>
@@ -1594,7 +1594,7 @@
                <item row="23" column="0">
                 <widget class="QLabel" name="peerAddrRelayEnabledLabel">
                  <property name="toolTip">
-                  <string>Whether we relay addresses to this peer.</string>
+                  <string extracomment="Tooltip text for the Address Relay field in the peer details area.">Whether we relay addresses to this peer.</string>
                  </property>
                  <property name="text">
                   <string>Address Relay</string>
@@ -1620,7 +1620,7 @@
                <item row="24" column="0">
                 <widget class="QLabel" name="peerAddrProcessedLabel">
                  <property name="toolTip">
-                  <string>Total number of addresses processed, excluding those dropped due to rate limiting.</string>
+                  <string extracomment="Tooltip text for the Addresses Processed field in the peer details area.">Total number of addresses processed, excluding those dropped due to rate-limiting.</string>
                  </property>
                  <property name="text">
                   <string>Addresses Processed</string>
@@ -1646,7 +1646,7 @@
                <item row="25" column="0">
                 <widget class="QLabel" name="peerAddrRateLimitedLabel">
                  <property name="toolTip">
-                  <string>Total number of addresses dropped due to rate limiting.</string>
+                  <string extracomment="Tooltip text for the Addresses Rate-Limited field in the peer details area.">Total number of addresses dropped due to rate-limiting.</string>
                  </property>
                  <property name="text">
                   <string>Addresses Rate-Limited</string>
```
</p></details>

**Review by hebasto — APPROVED** [review]
ACK 9fbd1bb7fabec76c668a649967f6f4e355e6ca59, tested on Ubuntu 21.10 (Qt 5.15.2).

**Review by w0xlt — APPROVED** [review]
reACK 9fbd1bb

File: src/qt/forms/debugwindow.ui [hunk 2]
```diff
@@ -1592,6 +1592,84 @@
...
+                 </property>
+                </widget>
+               </item>
+               <item row="24" column="0">
+                <widget class="QLabel" name="peerAddrProcessedLabel">
+                 <property name="toolTip">
+                  <string extracomment="Tooltip text for the Addresses Processed field in the peer details area.">Total number of addresses processed, excluding those dropped due to rate-limiting.</string>
+                 </property>
+                 <property name="text">
+                  <string>Addresses Processed</string>
```
Comment (hebasto) [inline src/qt/forms/debugwindow.ui:1626]: A [feedback](https://www.transifex.com/bitcoin/bitcoin/translate/#fr/qt-translation-023x/410183859) from Transifex translator:
> Rather: "Processed Addresses"

File: src/qt/forms/debugwindow.ui [hunk 3]
```diff
@@ -1592,6 +1592,84 @@
...
+                 </property>
+                </widget>
+               </item>
+               <item row="25" column="0">
+                <widget class="QLabel" name="peerAddrRateLimitedLabel">
+                 <property name="toolTip">
+                  <string extracomment="Tooltip text for the Addresses Rate-Limited field in the peer details area.">Total number of addresses dropped due to rate-limiting.</string>
+                 </property>
+                 <property name="text">
+                  <string>Addresses Rate-Limited</string>
```
Comment (hebasto) [inline src/qt/forms/debugwindow.ui:1652]: A [feedback](https://www.transifex.com/bitcoin/bitcoin/translate/#fr/qt-translation-023x/410183861) from Transifex translator:
> Rather: "Rate-Limited Addresses"

**Review by hebasto — COMMENTED** [review]
Cross-posted feedback from French translation team.

File: src/qt/forms/debugwindow.ui (hunk 2 above)
Comment (jonatack) [reply src/qt/forms/debugwindow.ui:1626]: If "processed" was an adjective, then I would agree with the feedback, i.e. "red car" and not "car red".

However, in this case it is a verb and this is fine as-is, i.e. "cars sold" vs "sold cars".

File: src/qt/forms/debugwindow.ui (hunk 3 above)
Comment (jonatack) [reply src/qt/forms/debugwindow.ui:1652]: Same response as https://github.com/bitcoin-core/gui/pull/526#discussion_r808822625.

File: src/qt/forms/debugwindow.ui (hunk 2 above)
Comment (AO-LocLab) [reply src/qt/forms/debugwindow.ui:1626]: Hi @jonatack The lack of string context in Transifex is a real problem for this project, as proper context (dev notes and/or screenshots) are essential to achieve quality translations.
How is this string used, is this a title, a tooltip, a notification? Do we actually mean “Addresses are processed”?

Comment (jonatack) [reply src/qt/forms/debugwindow.ui:1626]: @AO-LocLab it is the field name in the peer details area, as displayed in the screenshot in the PR description. There is also a tooltip saying this, and its explanatory description, 3 lines higher at line 1623. I don't know if these are visible in transifex.

```html
<string extracomment="Tooltip text for the Addresses Processed field in the peer details area.">
    Total number of addresses processed, excluding those dropped due to rate-limiting.</string>
```

Comment (hebasto) [reply src/qt/forms/debugwindow.ui:1626]: > I don't know if they are visible in transifex.

Screenshot from 2022-02-20 19-09-24

Comment (hebasto) [reply src/qt/forms/debugwindow.ui:1626]: Although, this one has no developer notes:

Screenshot from 2022-02-20 19-12-06

Comment (jonatack) [reply src/qt/forms/debugwindow.ui:1626]: > Although, this one has no developer notes:

Should an extracomment be added, or something else?

Comment (hebasto) [reply src/qt/forms/debugwindow.ui:1626]: > > Although, this one has no developer notes:
> 
> Should an extracomment be added, or something else?

Actually, `extracomment`s in `*.ui` files _are_ Qt [translator comments](https://doc.qt.io/qt-5/i18n-source-translation.html#translator-comments) which are presented on Transifex as "developer notes".

Comment (hebasto) [reply src/qt/forms/debugwindow.ui:1626]: > Should an extracomment be added, or something else?

That is the plan :)

Comment (jonatack) [reply src/qt/forms/debugwindow.ui:1626]: Ok! Will propose adding extracomments to the titles, then (if I understand correctly) and maybe to the tooltips without them.

Comment (hebasto) [reply src/qt/forms/debugwindow.ui:1626]: > Ok! Will propose adding extracomments to the titles, then (if I understand correctly) and maybe to the tooltips without them.

Thank you!

Comment (jonatack) [reply src/qt/forms/debugwindow.ui:1626]: Started adding and improving the dev notes (and tooltips) in #554.

**Pull request merged**